src/library_dylink.js

```diff
@@ -896,6 +896,37 @@ var LibraryDylink = {
     return dso;
   },
 
+  $isValidFile__proxy: 'sync',
+  $isValidFile__deps: ['$FS'],
+  $isValidFile: function(filenamePtr) {
+    var filename = UTF8ToString(filenamePtr);
+#if DYLINK_DEBUG
+    dbg('isValidFile: ' + filename);
+#endif
+    var target = FS.findObject(filename);
+    return target && !target.isFolder && !target.isDevice;
+  },
```

Comment:
Would it be better to check if the file is valid in `readFileProxied` (where it also does `FS.findObject`), and to return an error code if needed from there?

Reply:
I don't think so since we call `isValidFile` in a loop while searching LD_LIBRARY_PATH.